posthog/hogql/printer.py

```diff
@@ -0,0 +1,213 @@
+            elif node.name == "count":
+                response = "count(*)"
+            elif node.name == "countDistinct":
+                response = f"count(distinct {translated_args})"
+            elif node.name == "countDistinctIf":
+                response = f"countIf(distinct {translated_args})"
+            else:
+                response = f"{node.name}({translated_args})"
+
+        elif node.name in CLICKHOUSE_FUNCTIONS:
+            response = f"{CLICKHOUSE_FUNCTIONS[node.name]}({', '.join([print_ast(arg, stack, context, dialect) for arg in node.args])})"
+        else:
+            raise ValueError(f"Unsupported function call '{node.name}(...)'")
+    elif isinstance(node, ast.Placeholder):
+        raise ValueError(f"Found a Placeholder {{{node.field}}} in the tree. Can't generate query!")
+    else:
+        raise ValueError(f"Unknown AST node {type(node).__name__}")
+
+    stack.pop()
+    return response
+
+
+def parse_field_access(chain: List[str], context: HogQLContext) -> HogQLFieldAccess:
+    # Circular import otherwise
+    from posthog.models.property.util import get_property_string_expr
+
+    """Given a list like ['properties', '$browser'] or ['uuid'], translate to the correct ClickHouse expr."""
+    if len(chain) == 2:
+        if chain[0] == "properties":
+            key = f"hogql_val_{len(context.values)}"
+            context.values[key] = chain[1]
+            escaped_key = f"%({key})s"
+            expression, _ = get_property_string_expr(
+                "events",
+                chain[1],
+                escaped_key,
+                "properties",
+            )
+            return HogQLFieldAccess(chain, "event.properties", chain[1], expression)
+        elif chain[0] == "person":
+            if chain[1] in EVENT_PERSON_FIELDS:
+                return HogQLFieldAccess(chain, "person", chain[1], f"person_{chain[1]}")
+            else:
+                raise ValueError(f"Unknown person field '{chain[1]}'")
+    elif len(chain) == 3 and chain[0] == "person" and chain[1] == "properties":
+        key = f"hogql_val_{len(context.values or {})}"
+        context.values[key] = chain[2]
+        escaped_key = f"%({key})s"
+
+        if context.using_person_on_events:
+            expression, _ = get_property_string_expr(
+                "events",
+                chain[2],
+                escaped_key,
+                "person_properties",
+                materialised_table_column="person_properties",
+            )
+
+        else:
+            expression, _ = get_property_string_expr(
+                "person",
+                chain[2],
+                escaped_key,
+                "person_props",
+                materialised_table_column="properties",
+            )
+
+        return HogQLFieldAccess(chain, "person.properties", chain[2], expression)
+    elif len(chain) == 1:
+        if chain[0] in EVENT_FIELDS:
+            if chain[0] == "id":
```

Comment:
I see now that my previous comment is addressed here. id --> uuid is a magic mapping

Reply:
This can also be removed.